rust/numaflow-core/src/config/pipeline.rs

```diff
@@ -561,6 +598,26 @@ impl PipelineConfig {
             });
         }
 
+        let isb_config =
+            vertex_obj
+                .spec
+                .inter_step_buffer
+                .as_ref()
+                .map(|isb_spec| isb_config::ISBConfig {
+                    compression: isb_config::Compression {
+                        compress_type: match isb_spec.compression.as_ref() {
+                            None => isb_config::CompressionType::None,
+                            Some(t) => match &t.r#type {
+                                None => isb_config::CompressionType::None,
+                                Some(t) => match t.as_str() {
+                                    "gzip" => isb_config::CompressionType::Gzip,
+                                    _ => isb_config::CompressionType::None,
```

Comment:
This should be an error. Since it is a some variant, user intended to set a compression type, but an unsupported value was set.